common/network-types/src/frame.rs

```diff
@@ -202,7 +217,7 @@ impl<'a> FrameBuilder<'a> {
 
     /// Adds a new [`segment`](Segment) to the builder with a timestamp `ts`.
     /// Returns the number of segments remaining in this builder.
-    fn put(&self, segment: Segment<'a>, ts: u64) -> crate::errors::Result<u16> {
+    fn put(&self, segment: Segment, ts: u64) -> crate::errors::Result<u16> {
```

Comment:
Wouldn't
```suggestion
    fn put(&self, segment: Segment, ts: std::time::Duration) -> crate::errors::Result<u16> {
```

be more idiomatic?

Reply:
https://rust-docs.redwoodjs.cn/std/sync/atomic/struct.AtomicU128.html

That requires native support though.

Reply:
`AtomicU64` shall be sufficient for millisecond precision.